packages/dapp/src/components/Listings.tsx

```diff
@@ -21,47 +31,119 @@ class Listings extends React.Component<{}, ListingsState> {
     super(props);
     this.state = {
       applications: List<string>(),
-      applicationSubscription: new Subscription(),
+      whitelistedListings: List<string>(),
+      readyToWhitelistListings: List<string>(),
+      inChallengeCommitListings: List<string>(),
+      inChallengeRevealListings: List<string>(),
+      canBeUpdatedListings: List<string>(),
+      awaitingAppealRequestListings: List<string>(),
+      awaitingAppealJudgmentListings: List<string>(),
+      compositeSubscription: new Subscription(),
       error: undefined,
     };
   }
 
-  public componentWillMount(): void {
-    window.addEventListener("load", this.initListings);
+  public async componentWillMount(): Promise<void> {
+    return this.initListings();
   }
 
   public componentWillUnmount(): void {
-    this.state.applicationSubscription.unsubscribe();
-    window.removeEventListener("load", this.initListings);
+    this.state.compositeSubscription.unsubscribe();
   }
 
   public render(): JSX.Element {
     return (
       <StyledDiv>
-        applications: {this.state.applications.toString()}
+        Whitelisted Newsrooms:<br />
+        <ListingList listings={this.state.whitelistedListings} />
+        <br />
+        Applications:<br />
+        <ListingList listings={this.state.applications} />
+        <br />
+        Ready to be Whitelisted:<br />
+        <ListingList listings={this.state.readyToWhitelistListings} />
+        <br />
+        In Challenge Vote-Commit Stage:<br />
+        <ListingList listings={this.state.inChallengeCommitListings} />
+        <br />
+        In Challenge Vote-Reveal Stage:<br />
+        <ListingList listings={this.state.inChallengeRevealListings} />
+        <br />
+        Can be Updated:<br />
+        <ListingList listings={this.state.canBeUpdatedListings} />
+        <br />
+        Awaiting Appeal Request:<br />
+        <ListingList listings={this.state.awaitingAppealRequestListings} />
+        <br />
+        Awaiting Appeal Judgment:<br />
+        <ListingList listings={this.state.awaitingAppealJudgmentListings} />
         <br />
         {this.state.error}
       </StyledDiv>
     );
   }
 
+  // TODO(nickreynolds): move this all into redux
   private initListings = async () => {
-    const civil = new Civil();
-    let tcr;
-    try {
-      tcr = civil.tcrSingletonTrusted();
-    } catch (ex) {
-      console.log("failed to get tcr.");
-      this.setState({
-        error: "No Supported Network Found. Please set MetaMask network to Rinkeby and Unlock Account.",
-      });
-    }
+    const tcr = getTCR();
 
     if (tcr) {
-      const subscription = tcr.listingsInApplicationStage().subscribe(listing => {
-        this.setState({ applications: this.state.applications.push(listing) });
-      });
-      this.setState({ applicationSubscription: subscription });
+      this.state.compositeSubscription.add(
```

Comment:
feel like there should be  a better way

Reply:
this subscription stuff is all going to change